### ISSUE TITLE
Indent rule not consistent in expression

### ISSUE THREAD

**Issue by rontav** [eslint/eslint#13854]
**Environment Info:**

Node version: v12.18.2
npm version: v6.14.5
Local ESLint version: v7.13.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

@babel/eslint-parser 7.12.1

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended"
    ],
    "parser": "@babel/eslint-parser",
    "parserOptions": {
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "rules": {
        "operator-linebreak": [
            "error",
            "after"
        ],
        "indent": [
            "error",
            4
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let a, b, c, d, e, f, g;

if (a && (
    b &&
c && (
        d ||
                        e ||
f
    ) &&
g
)) {
    console.log();
}
```

```bash
npx eslint index.js
```

**What did you expect to happen?**

It should show an error on line 5, 7, 8 and 10.
I expected the following indentations:
- line 5: 4 spaces
- line 7: 8 spaces
- line 8: 8 spaces
- line 10: 4 spaces

**What actually happened? Please include the actual, raw output from ESLint.**

No output (no errors).

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by rontav** [comment]
Only the first line inside of an expression inside parentheses seems to be analysed by the indentation rule.

**Comment by btmills** [comment]
[Demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNF0gKi9cblxubGV0IGEsIGIsIGMsIGQsIGUsIGYsIGc7XG5cbmlmIChhICYmIChcbiAgICBiICYmXG5jICYmICggLy8gc2hvdWxkIGJlIDQgc3BhY2VzXG4gICAgICAgIGQgfHxcbiAgICAgICAgICAgICAgICAgICAgICAgIGUgfHwgLy8gc2hvdWxkIGJlIDggc3BhY2VzXG5mIC8vIHNob3VsZCBiZSA4IHNwYWNlc1xuICAgICkgJiZcbmcgLy8gc2hvdWxkIGJlIDQgc3BhY2VzXG4pKSB7XG4gICAgY29uc29sZS5sb2coKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) for anyone else who wants to take a look. This repros with espree, not just `@babel/eslint-parser`.

Thanks for the report, @rontav. As you might be able to guess because it has its own label, `indent` is probably our most complicated rule. Without digging into the cause, it looks like we might only be checking indentation at node boundaries rather than at newlines. I'll give the rest of the team a chance to see how easy this might be to fix.
